src/compiler/compile/render_dom/wrappers/shared/is_head.ts

```diff
@@ -1,3 +1,9 @@
-export function is_head(node) {
-	return node && node.type === 'MemberExpression' && node.object.name === '@_document' && node.property.name === 'head';
+import { Identifier, Node } from 'estree';
+
+type NarrowNodeType<Y> = Y extends { name: string } ? Y : never;
+
+export function is_head(node: Identifier | Node) {
```

Comment:
[imo]

```suggestion
export function is_head(node: Node) {
```

Just using `Node` is enough I think.
Because the below code has no compile error.

```ts
const i: Identifier = {type: 'Identifier', name: 'i'};
const n: Node = i;
```